### DISCUSSION TITLE
Is there a way to just use xgettext instead of the Clang plugin?

### DISCUSSION THREAD

**Discussion by viniciusferrao** [zauguin/i18n-cpp#6] [Q&A]
In a system where is impractical to use `clang`, like servers, there is a way to use `gcc` instead and the old `xgettext` binary to generate the translation files?

**Comment by zauguin** [comment]
As far as I can tell `xgettext` can't deal with user defined literals (which was one of the original motivations to this in the first place) and even if it did the plural handling would probably require quite some postprocessing...

It's not really a full solution, but if you just want to avoid having to use clang as a main compiler and having to load a plugin, you can try the (undocumented) `i18n-extract` tool. It uses clang internally, so it still requires your code to be compatible with clang and it might not be as fast as a tool which doesn't but if it's linked against a static version of libclang-cpp it should work even on systems with no clang installed.

It is build next to the plugin in `clang/i18n-extract`. You can find basic instructions by running it with the `--help` option. Basically you just pass a number of source files and then it generates the corresponding `.poc` files from them as if you would have compiled with the plugin.
